src/SpoutReader.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+namespace Port\Spout;
+
+use Box\Spout\Common\Type;
+use Box\Spout\Reader\ReaderFactory;
+use OutOfBoundsException;
+use Port\Exception\ReaderException;
+use Port\Reader\CountableReader;
+use SeekableIterator;
+
+
+/**
+ * Reads Excel files with the help of Spout
+ *
+ * Spout must be installed.
+ *
+ * @link http://opensource.box.com/spout/
+ * @link https://github.com/box/spout
+ */
+class SpoutReader implements CountableReader, SeekableIterator
+{
+    /**
+     * @var \Box\Spout\Reader\XLSX\Sheet
+     */
+    protected $sheet;
+
+    /**
+     * @var integer
+     */
+    protected $headerRowNumber;
+
+    /**
+     * @var array
+     */
+    protected $columnHeaders;
+
+    /**
+     * Total number of rows
+     *
+     * @var integer
+     */
+    protected $count;
+
+    /**
+     * @param \SplFileObject $file Excel file
+     * @param int $headerRowNumber Optional number of header row
+     * @param int $activeSheet Index of active sheet to read from
+     * @param bool $shouldPreserveEmptyRows Sets whether empty rows should be returned or skipped
+     *
+     * @throws ReaderException
+     */
+    public function __construct(\SplFileObject $file, $headerRowNumber = null, $activeSheet = null, $shouldPreserveEmptyRows = true)
+    {
+        $reader = $this->createReaderForFile($file, $shouldPreserveEmptyRows);
```

Comment:
I would rather inject the [`ReaderInterface`](https://github.com/box/spout/blob/master/src/Spout/Reader/ReaderInterface.php) directly.

Reply:
Fixed, the user must call the `ReaderInterface::open` method before use the `SpoutReader`

Reply:
With this fix, the `SpoutReader` is compatible with CSV, XLSX and ODS. 

But the method `count` of the ODS format throw an exception because the `RowIterator` can't be rewind more than once. See https://github.com/box/spout/blob/3681a3421a868ab9a65da156c554f756541f452b/src/Spout/Reader/ODS/RowIterator.php#L101

Maybe, I will remove the implementation of `CountableReader`